certification/internal/policy/operator/deployable_by_olm.go

```diff
@@ -432,112 +471,76 @@ func (p *DeployableByOlmCheck) installedCSV(ctx context.Context, operatorData Op
 		indexImageNamespace := strings.Split(operatorData.CatalogImage, "/")[1]
 		operatorServiceAccount := operatorData.App
 		operatorNamespace := operatorData.InstallNamespace
 		// remove pipeline-related rolebindings
-		p.OpenshiftEngine.DeleteRoleBinding(ctx, fmt.Sprintf("%s:%s:%s", pipelineServiceAccount, operatorNamespace, registryViewerRole), indexImageNamespace)
-		p.OpenshiftEngine.DeleteRoleBinding(ctx, fmt.Sprintf("%s:%s:%s", pipelineServiceAccount, operatorNamespace, imagePullerRole), indexImageNamespace)
+		p.openshiftEngine.DeleteRoleBinding(ctx, fmt.Sprintf("%s:%s:%s", pipelineServiceAccount, operatorNamespace, registryViewerRole), indexImageNamespace)
+		p.openshiftEngine.DeleteRoleBinding(ctx, fmt.Sprintf("%s:%s:%s", pipelineServiceAccount, operatorNamespace, imagePullerRole), indexImageNamespace)
 		// remove rolebindings required for the default OperatorHub catalog sources
-		p.OpenshiftEngine.DeleteRoleBinding(ctx, fmt.Sprintf("%s:%s:%s", operatorServiceAccount, openshiftMarketplaceNamespace, registryViewerRole), indexImageNamespace)
-		p.OpenshiftEngine.DeleteRoleBinding(ctx, fmt.Sprintf("%s:%s:%s", operatorServiceAccount, openshiftMarketplaceNamespace, imagePullerRole), indexImageNamespace)
+		p.openshiftEngine.DeleteRoleBinding(ctx, fmt.Sprintf("%s:%s:%s", operatorServiceAccount, openshiftMarketplaceNamespace, registryViewerRole), indexImageNamespace)
+		p.openshiftEngine.DeleteRoleBinding(ctx, fmt.Sprintf("%s:%s:%s", operatorServiceAccount, openshiftMarketplaceNamespace, imagePullerRole), indexImageNamespace)
 		// remove rolebindings required for custom catalog sources
-		p.OpenshiftEngine.DeleteRoleBinding(ctx, fmt.Sprintf("%s:%s:%s", operatorServiceAccount, operatorNamespace, registryViewerRole), indexImageNamespace)
-		p.OpenshiftEngine.DeleteRoleBinding(ctx, fmt.Sprintf("%s:%s:%s", operatorServiceAccount, operatorNamespace, imagePullerRole), indexImageNamespace)
+		p.openshiftEngine.DeleteRoleBinding(ctx, fmt.Sprintf("%s:%s:%s", operatorServiceAccount, operatorNamespace, registryViewerRole), indexImageNamespace)
+		p.openshiftEngine.DeleteRoleBinding(ctx, fmt.Sprintf("%s:%s:%s", operatorServiceAccount, operatorNamespace, imagePullerRole), indexImageNamespace)
 	}
-	p.OpenshiftEngine.DeleteNamespace(ctx, operatorData.InstallNamespace)
-	p.OpenshiftEngine.DeleteNamespace(ctx, operatorData.TargetNamespace)
+	p.openshiftEngine.DeleteNamespace(ctx, operatorData.InstallNamespace)
+	p.openshiftEngine.DeleteNamespace(ctx, operatorData.TargetNamespace)
 }
 
-func (p *DeployableByOlmCheck) writeToFile(data interface{}) error {
-	obj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(data)
-	if err != nil {
-		log.Error("unable to convert the object to unstructured.Unstructured: ", err)
-		return err
-	}
-
-	u := &unstructured.Unstructured{Object: obj}
-	switch data.(type) {
-	case *operatorv1alpha1.CatalogSource:
-		u.SetGroupVersionKind(schema.GroupVersionKind{
```

Comment:
I feel like this was added in for some reason, not sure what it is. We should make sure that the files get written properly and are in the expected format as they were before. IE this change doesn't break connect or the pipeline.

Reply:
To me, it looks like the unstructured stuff was literally there to be able to unmarshal and then use u.Name() and u.Kind(). If that's all it was for, it's a lot of garbage for nothing.

Reply:
It looks like it's updating the `obj` as well, which made me wonder if the `obj` that is passed in does not have any of the GVK information attached to it.

Reply:
Looks like it's not writing them at all ATM. I'll figure out what's going on and update.

Reply:
I'm getting files written in my testing. They're JSON. Don't remember if they used to be or not. LMK what you find.

Reply:
I'm not getting files. I think I did in a previous iteration of this PR.

Rather, I seem to get _some_ files, but the data is incomplete (missing GVK), and the filenames are incomplete (missing kind).

Reply:
Yeah. I'll put it back the way it way.